pkg/maputil/maputil.go

```diff
@@ -187,5 +188,55 @@ func Set(m map[string]any, key []string, value string) {
 		m, key = getCursor(key[0]).getMap(m), key[1:]
 	}
 
-	getCursor(key[0]).set(m, value)
+	getCursor(key[0]).set(m, typedVal(value, stringBool))
+}
+
+func typedVal(val string, st bool) any {
+	if st {
+		return val
+	}
+
+	if strings.EqualFold(val, "true") {
+		return true
+	}
+
+	if strings.EqualFold(val, "false") {
+		return false
+	}
+
+	if strings.EqualFold(val, "null") {
+		return nil
+	}
+
+	if strings.EqualFold(val, "0") {
+		return int64(0)
+	}
+
+	// If this value does not start with zero, try parsing it to an int
+	if len(val) != 0 && val[0] != '0' {
+		if iv, err := strconv.ParseInt(val, 10, 64); err == nil {
+			return iv
+		}
+	}
+
+	return val
+}
+
+func MergeMaps(a, b map[string]interface{}) map[string]interface{} {
```

Comment:
please add some comments and unit tests for this function. @tunahansezen

Reply:
I've added a test and some comments to key points. In "helm" package, handled set-values with this method. I belive example in the test is sufficiently descriptive.